src/firewall/core/nftables.py

```diff
@@ -239,10 +239,26 @@ def __init__(self, fw):
         self._nft_handle.set_json_output(True)
 
     def nft_cmd(self, json_root):
+        if isinstance(json_root, list):
+            # The caller gave us the JSON in a list, so we can drop the
+            # reference to the memory. Unpack the list, and delete the entry.
+            l = json_root
+            (json_root,) = l
+            del l[0]
```

Comment:
I don't see any cases where `json_root` would be a list. It's always a dict.

This code also doesn't work. If json_root is a list then you can't assign `l` to a tuple.

```
>>> foo=[1,2,3,4,5]
>>> (blah,) = foo
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: too many values to unpack (expected 1)
```

Or `json_root` is a list of one element, then this does nothing.

Reply:
I think it works as intended. `(blah,) = foo` is tuple unpacking, and unpacks an interable of one element. It checks that the number of element matches, which is the reason for doing this instead of `blah = foo[0]`.

> Or json_root is a list of one element, then this does nothing.

If the caller passes `json_root`  as a list, then it MUST be a list of one element (otherwise, the tuple unpacking fails). But then it does the right thing. Unpack the list, and clear the element from the list.

> I don't see any cases where json_root would be a list. It's always a dict.

yes, it was previously unused. I now added a two commits
- improvement(nftables): avoid creating JSON string twice before nft_cmd()
- improvement(nftables): free json data earlier during nft_cmd()